grakn-engine/src/main/java/ai/grakn/engine/postprocessing/ConceptFixer.java

```diff
@@ -43,53 +51,122 @@
     private static final Logger LOG = LoggerFactory.getLogger(ConfigProperties.LOG_NAME_POSTPROCESSING_DEFAULT);
     private static final int MAX_RETRY = 10;
 
-    public static void checkCasting(EngineCacheImpl cache, String keyspace, String castingId){
+    public static void checkResources(String keyspace, String index, Set<ConceptId> conceptIds){
+        runPostProcessingJob(ConceptFixer::runResourceFix, keyspace, index, conceptIds);
+    }
+
+    public static void checkCastings(String keyspace, String index, Set<ConceptId> conceptIds){
+        runPostProcessingJob(ConceptFixer::runCastingFix, keyspace, index, conceptIds);
+    }
+
+    /**
+     * Main method which attempts to run all post processing jobs.
+     *
+     * @param postProcessor The post processing job.
+     *                      Either {@link ConceptFixer#runResourceFix(EngineGraknGraph, Set)} or
+     *                      {@link ConceptFixer#runCastingFix(EngineGraknGraph, Set)}.
+     *                      This then returns a function which will complete the job after going through validation
+     * @param keyspace The keyspace to post process against.
+     * @param conceptIndex The unique index of the concept which must exist at the end
+     * @param conceptIds The conceptIds which effectively need to be merged.
+     * @return true if the job was executed successfully, otherwise false.
+     */
+    private static boolean runPostProcessingJob(BiFunction<EngineGraknGraph, Set<ConceptId>, BiConsumer<EngineCache, String>> postProcessor,
```

Comment:
The currying is strong in this one! It might be better to create a custom `PostProcessor` interface with a single method for this case.